pisa/utils/matrix.py

```diff
@@ -129,16 +129,17 @@ def test_frob_psd(A):
     assert is_psd_after, "did not produce PSD matrix"
     assert np.isclose(xdist, actual_dist), "actual distance differs from expectation"
 
-if __name__ == '__main__':
+def test_matrix_random():
+    """Unit test producing a number of random matrices and checking if the
+    approximated matrix is indeed PSD.
+    """
+    import sys
     m_test = np.array([[1, -1], [2, 4]])
     test_frob_psd(m_test)
-    print('matrix before:')
-    print(m_test)
-    print('matrix after:')
-    print(fronebius_nearest_psd(m_test))
-    print('The result matrix is psd and the Frobenius norm matches the expectation!')
-    print('testing random matrices...')
     for i in range(100):
         m_test = np.random.randn(3, 3)
         test_frob_psd(m_test)
-    print('Test passed!')
+    sys.stdout.write('<< PASS : test_matrix_random >>\n')
```

Comment:
please import at the top of your file `from pisa.utils.log import logging, set_verbosity` and use `logging.info('<< PASS : test_matrix_random >>')` here, then `set_verbosity(1)` under `if __name__ == '__main__'`.

This allows for central (or fine-grained) control of what kinds of messages PISA emits to the user. Having `print` or `stdout`/`stderr` calls directly takes this away. We don't do a lot to control these messages right now, but having this everywhere might be quite useful e.g. when we get proper py.test integration, and we might simply want _no_ messages emitted (and then we can control that simply).

Reply:
I didn't know how exactly this was supposed to be done, so I copied from the one file where it was done wrongly! I fixed it in `jsons.py` as well now.

Reply:
That was a bad example to take from, unfortunately, as we (at least used to) have circular import issues with `log` importing `jsons` and `jsons` importing `log`. This might be resolved now, but that's why that file in particular was different from the convention we try to follow elsewhere in PISA.

I don't seem to be able to replicate the circular import error with your modifications to jsons.py, so maybe I fixed that issue somewhere along the way and just never realized it... (I'll do a little more testing with your branch here to be sure, though, that it's not a problem any longer).